packages/node/src/events/protocol-message/controller.ts

```diff
@@ -12,6 +12,17 @@ export default async function protocolMessageEventController(
   requestHandler: RequestHandler,
   nodeMsg: NodeMessageWrappedProtocolMessage
 ) {
+  // FIXME: Take into account the protocols sequence length. More
+  //        generally this if statement is trying to capture the notion:
+  //        "do not restart a protocol if a message is received from another
+  //         client mid-protocol execution (assume IO_WAIT is listening)"
+  //
+  //        An alternative might be that IO_WAIT is hooked onto toms event
+  //        listeniner inside the instructionExecutor that emits noise
+  //        when dispatchReceivedMessage receives a message for an in-progress
+  //        protocol execution.
+  if (nodeMsg.data.seq === 2) return;
```

Comment:
@IIIIllllIIIIllllIIIIllllIIIIllllIIIIll Question. Should there be some kind of way for `dispatchReceivedMessage` to pass messages into `IO_WAIT` callbacks specifically?